test-e2e/server.js

```diff
@@ -142,12 +142,11 @@ async function createTestServer(t) {
   const server = createServer({
     ...getManagerOptions('test server'),
     serverName: 'test server',
-    serverPublicBaseUrl: 'http://localhost:' + port,
     serverBearerToken: 'ignored',
   })
-  const serverAddress = await server.listen({ port })
+  await server.listen({ port })
   t.after(() => server.close())
-  return serverAddress
+  return `http://localhost:${port}`
```

Comment:
`server.listen()` returns _one_ of the addresses that fastify is listening on (on my Mac, the ipv6 address). We want to test access to the server based on `localhost` rather than the listening address.